test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/DependentAssemblyTest.cs

```diff
@@ -0,0 +1,138 @@
+                        {
+                            throw new NotImplementedException();
+                        }
+                    }
+                }
+                """;
+
+                string expectedGeneratedFileName = $"GeneratedFunctionMetadataProvider.g.cs";
+                string expectedOutput = """
+                // <auto-generated/>
+                using System;
+                using System.Collections.Generic;
+                using System.Collections.Immutable;
+                using System.Text.Json;
+                using System.Threading.Tasks;
+                using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+                using Microsoft.Extensions.DependencyInjection;
+                using Microsoft.Extensions.Hosting;
+
+                namespace Microsoft.Azure.Functions.Worker
+                {
+                    public class GeneratedFunctionMetadataProvider : IFunctionMetadataProvider
+                    {
+                        public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+                        {
+                            var metadataList = new List<IFunctionMetadata>();
+                            var Function0RawBindings = new List<string>();
+                            Function0RawBindings.Add(@"{""name"":""req"",""type"":""HttpTrigger"",""direction"":""In"",""authLevel"":""Anonymous"",""methods"":[""get"",""post""]}");
+                            Function0RawBindings.Add(@"{""name"":""$return"",""type"":""http"",""direction"":""Out""}");
+
+                            var Function0 = new DefaultFunctionMetadata
+                            {
+                                Language = "dotnet-isolated",
+                                Name = "HttpTriggerSimple",
+                                EntryPoint = "FunctionApp.HttpTriggerSimple.Run",
+                                RawBindings = Function0RawBindings,
+                                ScriptFile = "TestProject.dll"
+                            };
+                            metadataList.Add(Function0);
+                            var Function1RawBindings = new List<string>();
+                            Function1RawBindings.Add(@"{""name"":""req"",""type"":""HttpTrigger"",""direction"":""In"",""authLevel"":""Anonymous"",""methods"":[""get"",""post""]}");
+                            Function1RawBindings.Add(@"{""name"":""$return"",""type"":""http"",""direction"":""Out""}");
+                
+                            var Function1 = new DefaultFunctionMetadata
+                            {
+                                Language = "dotnet-isolated",
+                                Name = "DependencyFunc",
+                                EntryPoint = "DependentAssemblyWithFunctions.DependencyFunc.Run",
+                                RawBindings = Function1RawBindings,
+                                ScriptFile = "TestProject.dll"
```

Comment:
Is `ScriptFile` supposed to be `TestProject.dll` here? Wondering if it should be `DependentAssemblyWithFunctions.dll`

Can you compare what it is from the JSON generator?

Reply:
It is supposed to be `DependentAssemblyWithFunctions.dll` - fixing